app/v1/controllers/SubmitReturnController.scala

```diff
@@ -0,0 +1,94 @@
+ * limitations under the License.
+ */
+
+package v1.controllers
+
+import cats.data.EitherT
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, AnyContentAsJson, ControllerComponents}
+import play.mvc.Http.MimeTypes
+import utils._
+import v1.controllers.requestParsers.SubmitReturnRequestParser
+import v1.hateoas.AmendHateoasBodies
+import v1.models.errors._
+import v1.models.request.SubmitReturnRawData
+import v1.services._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class SubmitReturnController @Inject()(val authService: EnrolmentsAuthService,
+                                       requestParser: SubmitReturnRequestParser,
+                                       service: SubmitReturnService,
+                                       idGenerator: IdGenerator,
+                                       cc: ControllerComponents)
+                                      (implicit ec: ExecutionContext)
+  extends AuthorisedController(cc) with BaseController with Logging with AmendHateoasBodies {
+
+  implicit val endpointLogContext: EndpointLogContext = EndpointLogContext(
+    controllerName = "SubmitReturnController",
+    endpointName = "submitVatReturn"
+  )
+
+  def submitReturn(vrn: String): Action[JsValue] =
+    authorisedAction(vrn).async(parse.json) { implicit request =>
+
+      implicit val correlationId: String = idGenerator.generateCorrelationId
+      logger.info(message = s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] " +
+        s"Submitting Vat Return for VRN : $vrn with correlationId : $correlationId")
+
+      val rawData: SubmitReturnRawData = SubmitReturnRawData(
+        vrn,
+        AnyContentAsJson(request.body)
+      )
+
+      val result =
+        for {
+          parsedRequest <- EitherT.fromEither[Future](requestParser.parseRequest(rawData))
+          serviceResponse <- EitherT(service.submitReturn(parsedRequest))
+        } yield {
+
+          logger.info(
+            s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] - " +
+              s"Success response received with CorrelationId: ${serviceResponse.correlationId}")
+
+          Created
+            .withApiHeaders(serviceResponse.correlationId)
+            .as(MimeTypes.JSON)
+        }
+
+    result.leftMap { errorWrapper =>
+      val resCorrelationId = errorWrapper.correlationId
+      val result = errorResult(errorWrapper).withApiHeaders(resCorrelationId)
+      logger.warn(
+        s"[${endpointLogContext.controllerName}][${endpointLogContext.endpointName}] - " +
+          s"Error response received with CorrelationId: $resCorrelationId")
+
+      result
+    }.merge
+  }
+
+  private def errorResult(errorWrapper: ErrorWrapper) = {
+    (errorWrapper.error: @unchecked) match {
+      case VrnFormatError | PeriodKeyFormatError | RuleIncorrectOrEmptyBodyError | ValueFormatError |
```

Comment:
Should be MtdErrorWithCustomMessage(RuleIncorrectOrEmptyBodyError.code) and MtdErrorWithCustomMessage(ValueFormatError.code)